dubbo-remoting/dubbo-remoting-netty4/src/main/java/org/apache/dubbo/remoting/transport/netty4/NettyConfigOperator.java

```diff
@@ -0,0 +1,81 @@
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.dubbo.remoting.transport.netty4;
+
+import org.apache.dubbo.common.URL;
+import org.apache.dubbo.common.constants.CommonConstants;
+import org.apache.dubbo.remoting.Channel;
+import org.apache.dubbo.remoting.ChannelHandler;
+import org.apache.dubbo.remoting.Codec2;
+import org.apache.dubbo.remoting.api.pu.ChannelHandlerPretender;
+import org.apache.dubbo.remoting.api.pu.ChannelOperator;
+import org.apache.dubbo.remoting.api.pu.DefaultCodec;
+
+import java.util.List;
+
+public class NettyConfigOperator implements ChannelOperator {
+
+    public NettyConfigOperator(NettyChannel channel, ChannelHandler handler) {
+        this.channel = channel;
+        this.handler = handler;
+    }
+
+    @Override
+    public void configChannelHandler(List<ChannelHandler> handlerList) {
+        if(channel instanceof NettyChannel) {
+            URL url = channel.getUrl();
+            Codec2 codec2 = url.getOrDefaultFrameworkModel().getExtensionLoader(Codec2.class).
+                getExtension(url.getProtocol());
+            if (!(codec2 instanceof DefaultCodec)){
+                NettyCodecAdapter codec = new NettyCodecAdapter(codec2, channel.getUrl(), handler);
+                ((NettyChannel) channel).getNioChannel().pipeline().addLast(
+                    codec.getDecoder()
+                ).addLast(
+                    codec.getEncoder()
+                );
+            }
+
+            for (ChannelHandler handler: handlerList) {
+                if (handler instanceof ChannelHandlerPretender) {
+                    Object realHandler = ((ChannelHandlerPretender) handler).getRealHandler();
+                    if(realHandler instanceof io.netty.channel.ChannelHandler) {
+                        ((NettyChannel) channel).getNioChannel().pipeline().addLast(
+                            (io.netty.channel.ChannelHandler) realHandler
+                        );
+                    }
+                }
+            }
+
+            // todo distinguish between client and server channel
+            if( isClientSide(channel).equalsIgnoreCase(CommonConstants.CONSUMER)){
+                //todo config client channel handler
+            }else if (isClientSide(channel).equalsIgnoreCase(CommonConstants.PROVIDER)){
+                NettyServerHandler sh = new NettyServerHandler(channel.getUrl(), handler);
+                ((NettyChannel) channel).getNioChannel().pipeline().addLast(
+                    sh
+                );
+            }
+        }
+    }
+
+    private String isClientSide(Channel channel) {
+        return channel.getUrl().getSide("");
+    }
```

Comment:
The method name should correspond to the method behavior, why not just judge here.